kubernetes-overview.el

```diff
@@ -412,7 +412,8 @@ SECTIONS is a list of sections to display.  See
      (list (alist-get choice views))))
 
   (kubernetes-state-update-overview-sections sections)
-  (kubernetes-state-trigger-redraw))
+  (kubernetes-state-trigger-redraw)
+  (run-hook-with-args 'kubernetes-poll-hook nil))
```

Comment:
> However, I don't agree w/ doing the same in `k8s-overview-set-sections`. I'd argue that in this case, the package is working as intended and simply obeying your very long poll frequency. For me, for example, I notice that switching from `overview` view (statefulsets+deployments) to `deployments` doesn't result in `Fetching...`, demonstrating that, when switching views, the package reuses the current state. This, to me, is desirable and intentional.
> 

I thought about this a lot, but there could be scenario were someone could have the following setting

```elisp
(setq kubernetes-default-overview-view 'deployments)
```

In that case only deployments will be polled ahead. User has to press `g` to load the data. 

Since by default value of the above variable is `overview` we are fetching many resources.

Reply:
> In that case only deployments will be polled ahead. User has to press `g` to load the data.

That's a good point. This suggests to me that the package should be intelligent about knowing when to short-circuit the polling loop and fetch unconditionally, e.g. in cases where there's nothing currently stored in state for the given resource. In other words, a sort of in-between compromise between the solution you propose with this PR and the behavior I describe as current and desirable.

Say, for example, you start with viewing Deployments only, via `(setq kubernetes-default-overview-view 'deployments)` (like you describe). Package will fetch Deployments and store them in state. Then, say the user switches to viewing some other resource, e.g. `pods`. The package would see that there's nothing stored in state for `pods` currently, and will then poll kubectl for pod info specifically, regardless of whether it's "ready" for polling according to the poll frequency.

What do you think of this proposed behavior?